File: source/SEP3 - test (Rider)/WebServices Unit Test/Model/Comment.cs
using System;

namespace Model
{
    public class Comment
    {
        public int GameId { get; set; }
        public string User { get; set; }
        public string Text { get; set; }
        public long? Time
        {
            get
            {
                if (TimeStamp == null)
                    return null;
                return (TimeStamp.Value.Ticks - new DateTime(1970, 1, 1).Ticks) / 10000;
            }
            set {if (value != null) TimeStamp = new DateTime(value.Value * 10000 + new DateTime(1970, 1, 1).Ticks);}
        }

        public DateTime? TimeStamp { get; set; }

        public override string ToString()
        {
            return "GameId: " + GameId + "\nUser: " + User + "\nText: " + Text + "\nTimeStamp: " + TimeStamp;
        }
    }
}

File: source/SEP3 - tier 3 (IntelliJ)/src/main/java/model/Comment.java
package model;

import java.util.Date;

/**this class serves as a Data Object for Comment*/
public class Comment
{

    /**id of the game which this comment is associated with*/
    public int GameId;
    /**name of the user which this comment is associated with*/
    public String User;
    /**content of the comment*/
    public String Text;
    /**datetime when the comment was posted (nullable)*/
    public Long Time;

    /**initialises all instance variables*/
    public Comment(int gameId, String user, String text, Long time)
    {
        GameId = gameId;
        User = user;
        Text = text;
        Time = time;
    }

    public String toString()
    {
        return GameId + " " + User + " " + Text + " " + new Date(Math.round(Time));
    }
}